src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -1969,18 +1961,44 @@ NEVER_INLINE EnumerateObjectData* ByteCodeInterpreter::executeEnumerateObject(Ex
     } else {
         size_t idx = 0;
         eData.idx = &idx;
-        data->m_keys.resizeWithUninitializedValues(ownKeyCount);
+        bool isNumber = true;
         target.asObject()->enumeration(state, [](ExecutionState& state, Object* self, const ObjectPropertyName& name, const ObjectStructurePropertyDescriptor& desc, void* data) -> bool {
+            EData* eData = (EData*)data;
             if (desc.isEnumerable()) {
-                EData* eData = (EData*)data;
-                eData->data->m_keys[(*eData->idx)++] = name.toPlainValue(state);
+                auto value = name.toPlainValue(state);
```

Comment:
Now, the below algorithm is similar to Object::ownPropertyKeys. What do you think? If you agree with my opinion, Could you remove the redundant codes?

Reply:
good idea! i removed the redundant codes